pkg/plugin/plugin.go

```diff
@@ -65,14 +71,22 @@ func assertOvsAvailable() error {
 	return nil
 }
 
-func loadNetConf(bytes []byte) (*netConf, error) {
-	netconf := &netConf{}
-	if err := json.Unmarshal(bytes, netconf); err != nil {
-		return nil, fmt.Errorf("failed to load netconf: %v", err)
+func getEnvArgs(envArgsString string) (*EnvArgs, error) {
+	if envArgsString != "" {
+		e := EnvArgs{}
+		err := types.LoadArgs(envArgsString, &e)
+		if err != nil {
+			return nil, err
+		}
+		return &e, nil
 	}
+	return nil, nil
+}
 
-	if netconf.BrName == "" {
-		return nil, fmt.Errorf("\"bridge\" is a required argument")
+func loadNetConf(bytes []byte, bridgeIsRequired bool) (*netConf, error) {
```

Comment:
`bridgeIsRequired` should not be passed down to this level. Also, we cannot ignore failure without checking that it was indeed because of missing bridge argument.

We should set BrName as `omitempty` and check if it was provided or not in caller of `loadNetConf`